Add --gen-inventory command to atomic-openshift-installer.

dgoodwin opened the pull request:
This will just generate the Ansible inventory once all information is gathered,
inform the user where to find it, and exit.

Includes some test utility updates to make what broke as a result of this
change less fragile.

Ping @smunilla

detiber commented on utils/src/ooinstall/cli_installer.py:
```diff
@@ -871,7 +872,16 @@ def install(ctx, force):
+        click.echo()
+        click.echo("Generated ansible inventory: %s" % inventory_file)
+        click.echo()
+        sys.exit(0)
```
Any objections to displaying the path to the generated inventory for all deployments and just having the new flag only generate the inventory?

dgoodwin replied:
Sounds good to me. I'll adjust tomorrow morn.

We should probably get it out of .ansible too someday.
On Mar 23, 2016 4:28 PM, "Jason DeTiberus" <notifications@github.com> wrote:

> In utils/src/ooinstall/cli_installer.py
> <https://github.com/openshift/openshift-ansible/pull/1654#discussion_r57222133>
> :
>
> >      oo_cfg.save_to_disk()
> > +    # Write ansible inventory file to disk:
> > +    inventory_file = openshift_ansible.generate_inventory(hosts_to_run_on)
> > +
> > +    if gen_inventory:
> > +        click.echo()
> > +        click.echo("Generated ansible inventory: %s" % inventory_file)
> > +        click.echo()
> > +        sys.exit(0)
>
> Any objections to displaying the path to the generated inventory for all
> deployments and just having the new flag only generate the inventory?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/openshift-ansible/pull/1654/files/9c90c5a6071d2bcc9a4cab449327cc1d44f17726#r57222133>
>

detiber replied:
:+1: on moving it out of .ansible

dgoodwin commented:
Modified to always write out the inventory location, slight text edits to make it consistent with similar line for the quick installer config write, added help for the new gen-inventory option.

Added another commit to move the inventory out of .ansible.

detiber commented:
aos-ci-test

detiber commented:
lgtm

brenton commented:
@dgoodwin, would you mind rebasing this?  For some reason whenever I try to merge it github reports that the merge failed.  It seems stuck in 'Checking for ability to automatically merge…' now.

dgoodwin commented:
@brenton not sure what happened here, I went to rebase, it looks like the commit is already in master so I think your merge attempt worked.

@andrewklau is the above meant for another PR?

andrewklau commented:
@dgoodwin hmm I left the comment on this commit https://github.com/openshift/openshift-ansible/commit/67fda43df08a260571eb9b1d7f1ecdab5e323c41 not sure why it ended up here too :/

dgoodwin commented:
Nice little github bug there.
